front/src/modules/trainschedule/components/Timetable/utils.ts

```diff
@@ -1,18 +1,29 @@
 import dayjs from 'dayjs';
 
 import type { Duration } from 'utils/duration';
+import { isPacedTrainWithDetails } from 'utils/trainId';
 
 import { specialCodeDictionary } from './consts';
 import type { TimetableItemWithDetails } from './types';
 
 /** Filter timetable items by their names and labels */
 export const keepItem = (item: TimetableItemWithDetails, searchString: string): boolean => {
   if (searchString) {
+    let hasMatchingExceptions = false;
+    if (isPacedTrainWithDetails(item)) {
+      hasMatchingExceptions = item.exceptions.some(
+        (exception) =>
+          exception.train_name?.value.toLowerCase().includes(searchString.toLowerCase()) ||
+          exception.labels?.value.some((label) =>
+            label.toLowerCase().includes(searchString.toLowerCase())
```

Comment:
Nit: Maybe you can extract those two test in specific and testable function,
one testing by the train_name the other by the label ?

Reply:
Yes sure that was nit :)